docs/scripts/api/buildComponentsDocumentation.ts

```diff
@@ -145,7 +146,12 @@ function extractSlots(options: {
       return;
     }
 
-    slots[name] = {
+    // Workaround to generate correct (camelCase) keys for slots in v6 `API Reference` documentation
+    // TODO v7: Remove this condition when `Grid(Pro|Premium)SlotsComponent` type is refactored to have `camelCase` names
+    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedGridSlotsComponent`
+    const slotName = project.name.includes('grid') ? camelCase(name) : name;
```

Comment:
Should we do something similar for pickers too as we deprecated `components/componentsProps` and used `slots/slotProps` instead in [demos](https://next.mui.com/x/react-date-pickers/date-picker/#helper-text) but in [API docs](https://next.mui.com/x/api/date-pickers/date-picker/#slots) we are still showing (PascalCase) keys (as per `components/componentsProps`) which may be confusing for users 🤔 

CC @alexfauquette